### DISCUSSION TITLE
Olimex ESP32-POE-ISO no connection to inverter when ethernet enabled

### DISCUSSION THREAD

**Discussion by quaternionma** [tbnobody/OpenDTU#908] [Q&A]
**My config:**
NRF24L01+ connected with the UEXT interface of Olimex ESP32
MISO=GPIO15
MOSI=GPIO13
SCLK=GPIO16
IRQ=GPIO2
CE=GPIO4
CS=GPIO5

![Bildschirmfoto vom 2023-05-10 18-22-02](https://github.com/tbnobody/OpenDTU/assets/11835492/3f8e27ce-d083-4883-b27e-aacf64f1c628)

This is the relevant portion of the device profile:

"name": "Olimex ESP32-POE",
        "nrf24": {
            "miso": 15,
            "mosi": 13,
            "clk": 16,
            "irq": 2,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": 12,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
         }

If the LAN cable is disconnected and the ESP32 connects through WIFiI the DTU can poll the data from the Inverter (HM-600)

`RX Period End
18:14:35.890 > Success
18:14:39.295 > Fetch inverter: 114184918154
18:14:39.366 > TX RealTimeRunData Channel: 75 --> 15 84 91 81 54 80 11 81 88 80 0B 00 64 5B C2 EF 00 00 00 00 00 00 00 00 62 D8 6E 
18:14:39.385 > Interrupt received
18:14:39.391 > RX Channel: 75 --> 95 84 91 81 54 84 91 81 54 01 00 01 01 26 00 6B 01 39 01 28 00 69 01 35 00 00 95 
18:14:39.419 > Interrupt received
18:14:39.426 > RX Channel: 3 --> 95 84 91 81 54 84 91 81 54 02 4B 11 00 00 4D 55 06 71 06 F3 09 25 13 86 02 53 BF 
18:14:39.465 > Interrupt received
18:14:39.472 > RX Channel: 3 --> 95 84 91 81 54 84 91 81 54 83 00 01 00 19 03 E8 01 18 00 54 6C 8C 48 
18:14:39.831 > RX Period End
18:14:39.831 > Success
18:14:40.493 > Admin mode disabled
18:14:44.300 > Fetch inverter: 114184918154
18:14:44.318 > TX RealTimeRunData Channel: 3 --> 15 84 91 81 54 80 11 81 88 80 0B 00 64 5B C2 F4 00 00 00 00 00 00 00 00 92 66 3B 
18:14:44.347 > Interrupt received
18:14:44.354 > RX Channel: 3 --> 95 84 91 81 54 84 91 81 54 01 00 01 01 26 00 6A 01 38 01 27 00 69 01 35 00 00 9A 
18:14:44.858 > RX Period End
18:14:44.858 > Last missing
18:14:44.858 > Request retransmit: 2
18:14:44.858 > TX RequestFrame Channel: 23 --> 15 84 91 81 54 80 11 81 88 82 CF 
18:14:44.865 > Interrupt received
18:14:44.873 > RX Channel: 3 --> 95 84 91 81 54 84 91 81 54 02 4B 11 00 00 4D 55 06 71 06 F3 09 23 13 87 02 51 BA 
18:14:44.980 > RX Period End
18:14:44.980 > Last missing
18:14:44.980 > Request retransmit: 3
18:14:44.980 > TX RequestFrame Channel: 40 --> 15 84 91 81 54 80 11 81 88 83 CE 
18:14:46.396 > Interrupt received
18:14:46.401 > RX Channel: 23 --> 95 84 91 81 54 84 91 81 54 83 00 00 00 19 03 E8 01 18 00 54 95 5A 66 
18:14:46.410 > RX Period End
18:14:46.410 > Success
18:14:49.302 > Fetch inverter: 114184918154
18:14:49.336 > TX RealTimeRunData Channel: 61 --> 15 84 91 81 54 80 11 81 88 80 0B 00 64 5B C2 F9 00 00 00 00 00 00 00 00 02 3E FE 
18:14:49.358 > Interrupt received
18:14:49.364 > RX Channel: 3 --> 95 84 91 81 54 84 91 81 54 01 00 01 01 26 00 6A 01 37 01 27 00 68 01 34 00 00 95 
18:14:49.405 > Interrupt received
18:14:49.412 > RX Channel: 23 --> 95 84 91 81 54 84 91 81 54 02 4B 11 00 00 4D 55 06 71 06 F3 09 26 13 87 02 50 BE 
18:14:49.465 > Interrupt received`

However if the ESP32 is connected via ethernet (via DHCP) the DTU gets in trouble getting the data from the inverter.
In both cases an external power supply is used, so no POE is activated

`RX Period End
18:33:56.304 > All missing
18:33:56.304 > Nothing received, resend count exeeded
18:33:56.347 > TX AlarmData Channel: 23 --> 15 84 91 81 54 80 11 81 88 80 11 00 64 5B C7 71 00 00 00 00 00 00 00 00 CE 33 A8 
18:33:56.994 > Websocket: [/livedata][8] disconnect
18:33:57.141 > RX Period End
18:33:57.141 > All missing
18:33:57.141 > Nothing received, resend whole request
18:33:57.141 > TX AlarmData Channel: 40 --> 15 84 91 81 54 80 11 81 88 80 11 00 64 5B C7 71 00 00 00 00 00 00 00 00 CE 33 A8 
18:33:57.933 > RX Period End
18:33:57.933 > All missing
18:33:57.933 > Nothing received, resend whole request
18:33:57.933 > TX AlarmData Channel: 61 --> 15 84 91 81 54 80 11 81 88 80 11 00 64 5B C7 71 00 00 00 00 00 00 00 00 CE 33 A8 
18:33:58.704 > RX Period End
18:33:58.704 > All missing
18:33:58.704 > Nothing received, resend whole request
18:33:58.704 > TX AlarmData Channel: 75 --> 15 84 91 81 54 80 11 81 88 80 11 00 64 5B C7 71 00 00 00 00 00 00 00 00 CE 33 A8 
18:33:59.497 > RX Period End
18:33:59.497 > All missing
18:33:59.497 > Nothing received, resend whole request
18:33:59.497 > TX AlarmData Channel: 3 --> 15 84 91 81 54 80 11 81 88 80 11 00 64 5B C7 71 00 00 00 00 00 00 00 00 CE 33 A8 
18:34:00.251 > RX Period End
18:34:00.251 > All missing
18:34:00.251 > Nothing received, resend count exeeded
18:34:00.251 > Fetch inverter: 114184918154
18:34:00.293 > TX RealTimeRunData Channel: 23 --> 15 84 91 81 54 80 11 81 88 80 0B 00 64 5B C7 78 00 00 00 00 00 00 00 00 44 42 40 
18:34:00.838 > RX Period End
18:34:00.838 > All missing
18:34:00.838 > Nothing received, resend whole request
18:34:00.838 > TX RealTimeRunData Channel: 40 --> 15 84 91 81 54 80 11 81 88 80 0B 00 64 5B C7 78 00 00 00 00 00 00 00 00 44 42 40 
18:34:01.382 > RX Period End
18:34:01.382 > All missing
18:34:01.382 > Nothing received, resend whole request
18:34:01.382 > TX RealTimeRunData Channel: 61 --> 15 84 91 81 54 80 11 81 88 80 0B 00 64 5B C7 78 00 00 00 00 00 00 00 00 44 42 40 
`

Any ideas?

**Comment by cablekai** [comment]
Same here,

with WiFi and USB-Power everything is fine.

Plugging in the Ethernet-cable (with or without POE) and the inverter "is gone"...

u can see in the logs that there are no interrupts received anymore...

:-( 

hope somebody has an idea

thx kai
